### PR TITLE
Add kvao delete

### PR DESCRIPTION
**Pull request by superkhau**
Delete a single key-value pair via key.

@bajtos PTAL

# Prerequisite of

- #1212

### CODE REVIEW

File: test/kvao/delete.suite.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const bdd = require('../helpers/bdd-if');
+const helpers = require('./_helpers');
+const should = require('should');
+
+module.exports = function(dataSourceFactory, connectorCapabilities) {
+  describe('delete', function() {
```
Comment (bajtos) [inline test/kvao/delete.suite.js]: This test suite should be disabled unless the connector supports the newly added feature, see https://github.com/strongloop/loopback-datasource-juggler/pull/1211#discussion_r94783632

File: test/kvao/delete.suite.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    function() {
+      return helpers.givenKeys(CacheItem, ['key1', 'key2'])
+        .then(() => CacheItem.delete('key1'))
+        .then(() => CacheItem.keys())
+        .then((keys) => {
+          should(keys).eql(['key2']);
+        });
+    });
+
+    it('does not remove key-value pairs unassociated to the given key',
```
Comment (bajtos) [inline test/kvao/delete.suite.js]: I wouldn't bother with this test. What kind of implementation would delete a single key across all models?

Comment (superkhau) [inline test/kvao/delete.suite.js]: true -- the point was to ensure deleting a single key does not have side effects that spill over to other models

**Comment by superkhau** [comment]
>Alternatively, it may be better to send a single pull request implementing all three parts - delete, optimized deleteAll and fall-back deleteAll.

Gonna do this instead, make it less confusing to figure out prerequisites, etc.

**Comment by superkhau** [comment]
Closing in favour of #1214

**Pull request closed without merging**